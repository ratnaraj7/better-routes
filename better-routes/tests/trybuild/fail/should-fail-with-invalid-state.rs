use axum::extract::State;
use axum_extra::routing::RouterExt;
use better_routes::{method_helper, routes};
use serde::Deserialize;
#[derive(Deserialize)]
struct Foo {
    id: String,
}
#[derive(Clone)]
struct FooState;
#[method_helper]
impl Foo {
    #[get]
    async fn handle_get(self, State(_): State<FooState>) {}
}
routes! {
    name => FooPath,
    "/:id" => Foo
}
fn main() {}
