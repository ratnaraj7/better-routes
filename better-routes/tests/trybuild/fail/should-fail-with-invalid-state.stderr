error[E0308]: mismatched types
  --> tests/trybuild/fail/should-fail-with-invalid-state.rs:13:13
   |
13 |     name => AllRoutes,
   |             ^^^^^^^^^
   |             |
   |             expected `Router`, found `Router<FooState>`
   |             expected `Router` because of return type
14 |     "/:id" => Foo {
15 |         get => get
   |                --- here the type of `r` is inferred to be `Router<FooState>`
   |
   = note: expected struct `Router<()>`
              found struct `Router<FooState>`
